File: Editorial/3_Longest_Substring_Without_Repeating_Characters/CSharp/1_Brute_Force_TLE.cs
public class Solution {
    public int LengthOfLongestSubstring(string s) {
        int n = s.Length;

        int res = 0;
        for (int i = 0; i < n; i++) {
            for (int j = i; j < n; j++) {
                if (CheckRepetition(s, i, j)) {
                    res = Math.Max(res, j - i + 1);
                }
            }
        }

        return res;
    }

    private bool CheckRepetition(string s, int start, int end) {
        HashSet<char> chars = new HashSet<char>();

        for (int i = start; i <= end; i++) {
            char c = s[i];
            if (chars.Contains(c)) {
                return false;
            }

            chars.Add(c);
        }

        return true;
    }
}

File: Editorial/3_Longest_Substring_Without_Repeating_Characters/TypeScript/1_Brute_Force_TLE.ts
function lengthOfLongestSubstring(s: string): number {
    const n = s.length;

    let res = 0;
    for (let i = 0; i < n; i++) {
        for (let j = i; j < n; j++) {
            if (checkRepetition(s, i, j)) {
                res = Math.max(res, j - i + 1);
            }
        }
    }

    function checkRepetition(s: string, start: number, end: number): boolean {
        const chars = new Set<string>();

        for (let i = start; i <= end; i++) {
            const c = s[i];
            if (chars.has(c)) {
                return false;
            }
            chars.add(c);
        }

        return true;
    }

    return res;
}
